### PR TITLE
examples: Add maxmod audio modes example

### PR DESCRIPTION
**Pull request by mdigiorgio**

### CODE REVIEW

File: examples/audio/maxmod/audio_modes/source/main.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identiier: CC0-1.0
+//
+// SPDX-FileContributor: Michele Di Giorgio, 2024
+
+// Example usage of the following audio modes:
+// - MM_MODE_A, Complete hardware mixing mode
+// - MM_MODE_B, Interpolated mixing mode
+// - MM_MODE_C, Extended mixing mode
+//
+// Check https://maxmod.org/ref/functions/mmSelectMode.html for more
```
Comment (AntonioND) [inline examples/audio/maxmod/audio_modes/source/main.c:10]: Eh... that's the devkitPro-maintained documentation 😅 You can link here instead: https://blocksds.github.io/docs/maxmod/group__nds__arm9__init.html

File: examples/audio/maxmod/basic_sounds/source/main.c [hunk 2]
```diff
@@ -28,7 +28,7 @@ int main(int argc, char **argv)
     mmLoad(MOD_PARALLAX_80599);
     mmLoad(MOD_LASSE_HAEN_PYYKIT);
 
-    printf("maxmod example\n");
+    printf("maxmod basic sounds example\n");
     printf("==============\n");
```
Comment (AntonioND) [inline examples/audio/maxmod/basic_sounds/source/main.c:32]: Extend the `====` line to match the text above.

**Comment by AntonioND** [comment]
Actually, I've just noticed that you have moved the other maxmod example to a subdirectory. If you're going to do this, move all the examples (including the `maxmod_nitrofs` example) and add a readme to the new folder (and update the one of the old folder!).

**Comment by mdigiorgio** [comment]
I've moved the directories one level up as requested. Plus, there is a new commit with another example of use of Maxmod

File: examples/maxmod/reverb/source/main.c [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+int initReverb()
+{
+    // Enable reverb system
+    mmReverbEnable();
+
+    // Allocate reverb memory for right and left channels
+    const size_t bufferLeftLen = mmReverbBufferSize(bitDepth,
+                                                    samplingRate,
+                                                    reverbDelayLeft);
+    uint8_t *bufferLeft = (uint8_t *)malloc(bufferLeftLen * 4);
```
Comment (AntonioND) [inline examples/maxmod/reverb/source/main.c:33]: You don't need the `(uint8_t *)` cast from malloc.

File: examples/maxmod/reverb/source/main.c [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    if (bufferLeft == NULL) {
+        printf("Not enough memory to allocate left channel buffer\n");
+        return 1;
+    }
+
+    const size_t bufferRightLen = mmReverbBufferSize(bitDepth,
+                                                     samplingRate,
+                                                     reverbDelayRight);
+    uint8_t *bufferRight = (uint8_t *)malloc(bufferRightLen * 4);
```
Comment (AntonioND) [inline examples/maxmod/reverb/source/main.c:42]: Same.

**Pull request merged**